### ISSUE TITLE
Custom Font

### ISSUE THREAD

**Issue by rkarimi88** [faruktoptas/FancyShowCaseView#109]
Hi
Thanks for your great lib.

Would you please add a function for changing typeface?
Of course I can use custom view, but if you add it it would save time :)

**Comment by faruktoptas** [comment]
Hi @rkarimi88 
You can set a title style:

```java
new FancyShowCaseView.Builder(this)
    .focusOn(view)
    .title("Background color and title style can be changed")
    .titleStyle(R.style.MyTitleStyle, Gravity.TOP | Gravity.CENTER)
    .build()
    .show();
```

```xml
<style name="MyTitleStyle" parent="@android:style/Widget.TextView">
    <item name="android:textSize">20sp</item>
    <item name="android:textColor">#00ff00</item>
    <item name="android:fontFamily">sans-serif-condensed</item>
</style>
```

If you use a library like [Calligraphy](https://github.com/chrisjenx/Calligraphy) you can set typefaces from styles.

**Comment by rkarimi88** [comment]
Thanks for your answer.
I want to use my own font, so i can't use font-family.

"If you use a library like Calligraphy you can set typefaces from styles."
Yes I'm using it, but i can't set it. i use:
<item name="fontPath">fonts/myfont.ttf</item>
but it doesn't work. the way i use is wrong?

**Comment by faruktoptas** [comment]
Did you add the code below to your activity?

```
@Override
protected void attachBaseContext(Context newBase) {
    super.attachBaseContext(CalligraphyContextWrapper.wrap(newBase));
}
```

**Comment by faruktoptas** [comment]
Can you share your code please?

**Comment by rkarimi88** [comment]
Of course:
in styles.xml:
`<style name="MyTitleStyle" parent="@android:style/Widget.TextView">
        <item name="android:textSize">20sp</item>
        <item name="android:textColor">#1babcf</item>
        <item name="fontPath">fonts/11.ttf</item>
    </style>`

where 11.ttf is a font in assets/fonts folder.

in java:
`FocusShape rect = FocusShape.ROUNDED_RECTANGLE;
FancyShowCaseView.Builder num1 = new FancyShowCaseView.Builder(this)
                .focusOn(Target)
                .title(getString(R.string.target_text))
                .titleSize((int) getResources().getDimension(R.dimen.MediumTextSize), TypedValue.COMPLEX_UNIT_PX)
                .titleStyle(R.style.MyTitleStyle, Gravity.TOP)
                .focusShape(rect);`

**Comment by faruktoptas** [comment]
Can you remove `titleSize()` method and try again?

**Comment by rkarimi88** [comment]
I did it, no difference.

**Comment by faruktoptas** [comment]
Where do you call show() method? In which method of activity/fragment?

**Comment by rkarimi88** [comment]
I'm using a queue and call `queue.show()` in onCreate method of Activity.

**Comment by faruktoptas** [comment]
I need to see whole code

**Comment by rkarimi88** [comment]
I'm using Android Studio 2.2.3 (old version) which I'm not able to update it till a few days later. Can the problem relates to this?
Your lib is very attractive and I really want to use it. I have tested many libs for showcase and yours is the best.

**Comment by faruktoptas** [comment]
It shouldn't be related with AS. It may be your device's Android version. Can you try with another device?

**Comment by rkarimi88** [comment]
I tried it. It's the same.
Here is my code:

styles.xml:
```xml
<resources>

    <!-- Base application theme. -->
    <style name="AppTheme" parent="Theme.AppCompat.Light.DarkActionBar">
        <!-- Customize your theme here. -->
        <item name="colorPrimary">@color/colorPrimary</item>
        <item name="colorPrimaryDark">@color/colorPrimaryDark</item>
        <item name="colorAccent">@color/colorAccent</item>
    </style>

    <style name="MyTitleStyle" parent="@android:style/Widget.TextView">
        <item name="android:textSize">20sp</item>
        <item name="android:textColor">#1babcf</item>
        <item name="fontPath">fonts/11.ttf</item>
    </style>

</resources>
```
MainActivity.java:
```java
public class MainActivity extends AppCompatActivity {

    //Font of Activity
    @Override
    protected void attachBaseContext(Context newBase) {
        super.attachBaseContext(CalligraphyContextWrapper.wrap(newBase));
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        TextView Target = (TextView) findViewById(R.id.Target);
        FocusShape rect = FocusShape.ROUNDED_RECTANGLE;
        FocusShape circ = FocusShape.CIRCLE;
        FancyShowCaseQueue queue = new FancyShowCaseQueue();

        FancyShowCaseView.Builder num1 = new FancyShowCaseView.Builder(this)
                .focusOn(Target)
                .title("This is description")
                //.titleSize((int) getResources().getDimension(R.dimen.MediumTextSize), TypedValue.COMPLEX_UNIT_PX)
                .titleStyle(R.style.MyTitleStyle, Gravity.TOP)
                .focusShape(rect);

        FancyShowCaseView.Builder num2 = new FancyShowCaseView.Builder(this)
                .focusOn(Target)
                .title("This is Second description")
                .titleGravity(Gravity.CENTER_HORIZONTAL)
                .focusShape(circ);

        queue.add(num1.build());
        queue.add(num2.build());
        queue.show();
    }
}
```
activity_main.xml:
```xml
<?xml version="1.0" encoding="utf-8"?>
<RelativeLayout xmlns:android="http://schemas.android.com/apk/res/android"
    xmlns:tools="http://schemas.android.com/tools"
    android:id="@+id/activity_main"
    android:layout_width="match_parent"
    android:layout_height="match_parent"
    android:paddingBottom="@dimen/activity_vertical_margin"
    android:paddingLeft="@dimen/activity_horizontal_margin"
    android:paddingRight="@dimen/activity_horizontal_margin"
    android:paddingTop="@dimen/activity_vertical_margin"
    tools:context="app.hrkarimi.mytest.MainActivity">

    <TextView
        android:id="@+id/Target"
        android:layout_width="wrap_content"
        android:layout_height="wrap_content"
        android:text="Hello World!" />
</RelativeLayout>
```

manifest.xml:
```xml
<?xml version="1.0" encoding="utf-8"?>
<manifest xmlns:android="http://schemas.android.com/apk/res/android"
    package="app.hrkarimi.mytest">

    <application
        android:name=".MyFont"
        android:allowBackup="true"
        android:icon="@mipmap/ic_launcher"
        android:label="@string/app_name"
        android:supportsRtl="true"
        android:theme="@style/AppTheme">
        <activity android:name=".MainActivity">
            <intent-filter>
                <action android:name="android.intent.action.MAIN" />

                <category android:name="android.intent.category.LAUNCHER" />
            </intent-filter>
        </activity>
    </application>

</manifest>
```

and at last, a class named MyFont.java for initializing Calligraphy:
```java
public class MyFont extends Application {

    @Override
    public void onCreate() {
        super.onCreate();
        CalligraphyConfig.initDefault(new CalligraphyConfig.Builder()
                .setDefaultFontPath("fonts/1.ttf")
                .setFontAttrId(R.attr.fontPath)
                .build()
        );
    }
}
```

**Comment by rkarimi88** [comment]
I apply `MyTitleStyle `to `Target `(TextView) in activity_main.xml and it works like a charm, but it doesn't work on ShowCase.

note that I have two fonts, named 1.ttf and 11.ttf, which 1.ttf is the font of whole activity, and i want 11.ttf to be the font of ShowCase. but none of them apply to ShowCase.
Even if 1.ttf would be applied to ShowCase, it was good... but it doesn't...

**Comment by faruktoptas** [comment]
FancyShowCaseView sets the style programmatically with setTextApperance() method. But I figured out that Calligraphy [doesn't support that](https://github.com/chrisjenx/Calligraphy/issues/309).
Unfortunately you have to use a custom view. 
I reopened this issue. Because I will add support for styling textview from xml.

**Comment by rkarimi88** [comment]
I'm so much thankful of you for spending time on this.

Thanks responsible man! :)

**Comment by faruktoptas** [comment]
Thanks @rkarimi88 :)
I'll let you know after fixing the issue. If you have time you can fix it and send me a pull request. You can contribute if you are interested in. I can help you 👍

**Comment by sabih1998** [comment]
I have the same problem...

**Comment by faruktoptas** [comment]
Fixed by setting Typeface to builder with #179
